### PR TITLE
ManiMo for Neural MP Release

### PR DESCRIPTION
**Pull request by Jim-Young6709**

### CODE REVIEW

File: manimo/actuators/arms/franka_arm.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 from manimo.actuators.arms.robot_ik.robot_ik_solver import RobotIKSolver
 
 # from manimo.actuators.arms.moma_arm import MujocoArmModel
-from manimo.actuators.controllers.policies import CartesianPDPolicy, JointPDPolicy
+from manimo.actuators.controllers.policies import CartesianPDPolicy, JointPDPolicy, OperationalSpaceLowFreq
```
Comment (mihdalal) [inline manimo/actuators/arms/franka_arm.py]: remove osc, not used in neural mp anyways

File: manimo/actuators/arms/franka_arm.py [hunk 2]
```diff
@@ -105,6 +105,7 @@ def reset(self):
         self._go_home()
         self.connect()
 
+        print("arm reset")
```
Comment (mihdalal) [inline manimo/actuators/arms/franka_arm.py]: delete

File: manimo/actuators/arms/franka_arm.py [hunk 3]
```diff
@@ -171,14 +198,22 @@ def _default_policy(self, action_space, kq_ratio=1.5, kqd_ratio=1.5):
...
+            )
 
     def _get_desired_pos_quat(self, eef_pose):
         if self.delta:
             ee_pos_cur, ee_quat_cur = self.robot.get_ee_pose()
             ee_pos_desired = ee_pos_cur + torch.Tensor(eef_pose[:3])
 
             # add two quaternions
-            ee_quat_desired = torch.Tensor(quat_add(ee_quat_cur, eef_pose[3:]))
+            ee_quat_desired = torch.Tensor(quat_add(eef_pose[3:], ee_quat_cur))
```
Comment (mihdalal) [inline manimo/actuators/arms/franka_arm.py]: double checking, is it intentional to change the order here, this doesn't seem correct

File: manimo/actuators/arms/franka_arm.py [hunk 4]
```diff
@@ -195,6 +230,7 @@ def _apply_joint_commands(self, q_desired):
                 {"joint_pos_desired": q_des_tensor.float()}
             )
         except grpc.RpcError:
+            print('grpc.RpcError in applying joint commands at franka_arm.py')
```
Comment (mihdalal) [inline manimo/actuators/arms/franka_arm.py]: unnecessary print

File: manimo/actuators/arms/franka_arm.py [hunk 5]
```diff
@@ -225,13 +261,35 @@ def _apply_eef_commands(self, eef_pose, wait_time=3):
...
         action_obs = {"delta": self.delta, "action": action.copy()}
-
-        # import pdb; pdb.set_trace()
         if self.action_space == ActionSpace.Cartesian:
             if self.ik_mode == IKMode.Polymetis:
-                self._apply_eef_commands(action)
+                self.robot.move_to_ee_pose(action[:3], action[3:], delta=False, time_to_go=None, op_space_interp=False)
+                desired_joint_action = torch.zeros(7)
+                ee_pos_desired = torch.zeros(3)
+                ee_quat_desired = torch.zeros(4)
```
Comment (mihdalal) [inline manimo/actuators/arms/franka_arm.py]: remove

File: manimo/conf/camera/multi_real_sense_depth_only.yaml
Comment (mihdalal) [inline manimo/conf/camera/multi_real_sense_depth_only.yaml:1]: what is this config used for

File: manimo/scripts/get_current_position.py [hunk 6]
```diff
@@ -13,10 +34,13 @@
...
+print(obs)
+cam = 'cam1c'
+print(obs['eef_pos'])
+depth = obs[f'{cam}_depth'][0]
+cv_depth = vis_depth(depth)
+cv2.imwrite('depth.png', cv_depth)
+cv2.imwrite('rgb.png', obs[f'{cam}'][0][:, :, ::-1])
+img = obs[f'{cam}'][0][:, :, ::-1]
+combined = cv2.addWeighted(img, 0.5, cv_depth, 0.5, 0)
+cv2.imwrite('combined.png', combined)
```
Comment (mihdalal) [inline manimo/scripts/get_current_position.py]: revert back to original

File: manimo/sensors/real_sense_camera.py [hunk 7]
```diff
@@ -49,17 +49,23 @@ def add_image(
             time.sleep(1.0 / camera_cfg.hz)
 
     align = rs.align(rs.stream.color)
+    depth_processing = (
+            rs.hole_filling_filter(),
+            rs.temporal_filter(),
+        )
```
Comment (mihdalal) [inline manimo/sensors/real_sense_camera.py]: put this under a flag but make it default true since our code uses this

File: manimo/actuators/arms/franka_arm.py (hunk 3 above)
Comment (Jim-Young6709) [reply manimo/actuators/arms/franka_arm.py]: not sure about the ordering. this change is probably from the original repo. I'll revert this then

File: manimo/conf/camera/multi_real_sense_depth_only.yaml
Comment (Jim-Young6709) [reply manimo/conf/camera/multi_real_sense_depth_only.yaml:1]: ah, this is used for close loop rollouts, disable rgb streaming to make it run faster. but... yeah, I think we should delete this for release

File: manimo/sensors/real_sense_camera.py (hunk 7 above)
Comment (Jim-Young6709) [reply manimo/sensors/real_sense_camera.py]: but what's the point of disabling this? I haven't seen any negative impact of this depth processing on the point cloud yet...

Comment (mihdalal) [reply manimo/sensors/real_sense_camera.py]: just cleaner to put it under a flag since its not standard to this code base. you can make the default true

File: manimo/actuators/arms/franka_arm.py [hunk 8]
```diff
@@ -41,10 +41,11 @@ def __init__(self, arm_cfg: DictConfig):
         self.ik_mode = IKMode(arm_cfg.ik_mode)
         self.JOINT_LIMIT_MIN = arm_cfg.joint_limit_min
         self.JOINT_LIMIT_MAX = arm_cfg.joint_limit_max
-        self.robot = CustomRobotInterace(
+        self.robot = RobotInterface(
             ip_address=self.config.robot_ip, enforce_version=False
         )
-        self.robot.hz = self.hz
+        if self.config.ik_mode == 'DMControl':
+            self.robot.hz = self.hz
```
Comment (mihdalal) [inline manimo/actuators/arms/franka_arm.py]: is this change necessary

File: manimo/actuators/arms/franka_arm.py [hunk 9]
```diff
@@ -133,13 +133,39 @@ def _go_home(self):
...
+            kqd=kqd,
+        )
+        self.robot.send_torch_policy(policy, blocking=False)
+        rate.sleep()
+        for joint_position in joint_positions:
+            self.robot.update_current_policy({"q_desired": joint_position})
+            rate.sleep()
+            
+        self.connect()
+
```
Comment (mihdalal) [inline manimo/actuators/arms/franka_arm.py:161]: is this necessary

File: manimo/actuators/arms/franka_arm.py [hunk 10]
```diff
@@ -133,13 +133,39 @@ def _go_home(self):
...
     def _default_policy(self, action_space, kq_ratio=1.5, kqd_ratio=1.5):
         q_initial = self.robot.get_joint_positions()
         kq = kq_ratio * torch.Tensor(self.kq)
         kqd = kqd_ratio * torch.Tensor(self.kqd)
         kx = torch.Tensor(self.robot.metadata.default_Kx)
         kxd = torch.Tensor(self.robot.metadata.default_Kxd)
-
```
Comment (mihdalal) [inline manimo/actuators/arms/franka_arm.py]: undo remove line

File: manimo/actuators/arms/franka_arm.py [hunk 11]
```diff
@@ -227,8 +253,6 @@ def _apply_eef_commands(self, eef_pose, wait_time=3):
 
     def step(self, action):
         action_obs = {"delta": self.delta, "action": action.copy()}
-
```
Comment (mihdalal) [inline manimo/actuators/arms/franka_arm.py:230]: undo remove line

File: manimo/actuators/arms/franka_arm.py (hunk 1 above)
Comment (Jim-Young6709) [reply manimo/actuators/arms/franka_arm.py]: fixed

File: manimo/actuators/arms/franka_arm.py (hunk 2 above)
Comment (Jim-Young6709) [reply manimo/actuators/arms/franka_arm.py]: fixed

File: manimo/actuators/arms/franka_arm.py (hunk 4 above)
Comment (Jim-Young6709) [reply manimo/actuators/arms/franka_arm.py]: fixed

File: manimo/actuators/arms/franka_arm.py (hunk 5 above)
Comment (Jim-Young6709) [reply manimo/actuators/arms/franka_arm.py]: removed

File: manimo/scripts/get_current_position.py (hunk 6 above)
Comment (Jim-Young6709) [reply manimo/scripts/get_current_position.py]: reverted

File: manimo/sensors/real_sense_camera.py (hunk 7 above)
Comment (Jim-Young6709) [reply manimo/sensors/real_sense_camera.py]: updated

File: manimo/actuators/arms/franka_arm.py (hunk 8 above)
Comment (Jim-Young6709) [reply manimo/actuators/arms/franka_arm.py]: reverted

File: manimo/actuators/arms/franka_arm.py (hunk 9 above)
Comment (Jim-Young6709) [reply manimo/actuators/arms/franka_arm.py:161]: yes, this is the soft control with lower control gains, used in reset and some other places

File: manimo/actuators/arms/franka_arm.py (hunk 10 above)
Comment (Jim-Young6709) [reply manimo/actuators/arms/franka_arm.py]: fixed

File: manimo/actuators/arms/franka_arm.py (hunk 11 above)
Comment (Jim-Young6709) [reply manimo/actuators/arms/franka_arm.py:230]: fixed
